### PR TITLE
Marcos.reginato/add fab

### PR DESCRIPTION
**Pull request by regmarcos**
Hi guys! In this PR i added three floating action buttons: A refresh button, that call the same service to give us the list of characters; a database refresh button, that gets the list of character that is already in the database; and finally the clear button that cleans the screen.
In the next gif, you can see how this three buttons works.

https://gfycat.com/highwastefulfulmar

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -30,6 +36,7 @@ dependencies {
     implementation project(':domain')
     implementation project(':data')
     implementation "androidx.cardview:cardview:1.0.0"
```
Comment (SartiMau) [inline app/build.gradle]: add these dependencies to the generalDependencies please

File: app/build.gradle [hunk 2]
```diff
@@ -52,6 +59,7 @@ dependencies {
             generalTestDependencies.kotlinTest
```
Comment (SartiMau) [inline app/build.gradle]: delete this unnecessary lines pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt [hunk 3]
```diff
@@ -28,19 +29,48 @@ class CharacterPresenter constructor(view: CharacterView,
...
+        requestGetCharacters()
+    }
+
+    fun onClickDatabaseFAB() {
+        view.hideFAB()
+        characters = emptyList()
+        view.showCharacters(characters)
+        view.showLoading()
+        characters = getCharacterRepositoryUseCase.invoke()
+        view.showCharacters(characters)
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt]: you can use directly 
view.showCharacters(getCharacterRepositoryUseCase.invoke())

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt [hunk 4]
```diff
@@ -28,19 +29,48 @@ class CharacterPresenter constructor(view: CharacterView,
...
+        view.showFAB()
+        view.hideLoading()
+    }
+
+    fun onClickClearFAB() {
+        characters = emptyList()
+        view.showCharacters(characters)
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt]: you can use directly
view.showCharacters(emptyList())

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -13,10 +14,43 @@
...
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/refresh_fab"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|end"
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:clickable="true"
+        android:focusable="true"
+        android:src="@android:drawable/stat_notify_sync"
+        app:backgroundTint="#353892"/>
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: use colors pls

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -13,10 +14,43 @@
...
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/database_fab"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|start"
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:clickable="true"
+        android:focusable="true"
+        android:src="@drawable/database_fab_icon"
+        app:backgroundTint="#353892" />
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: same

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -13,10 +14,43 @@
...
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/clear_fab"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|center"
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:clickable="true"
+        android:focusable="true"
+        android:src="@android:drawable/ic_menu_delete"
+        app:backgroundTint="#353892" />
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: same

File: buildsystem/dependencies.gradle [hunk 8]
```diff
@@ -26,29 +26,29 @@ ext {
 
 
     generalDependencies = [
-            rxAndroid : "io.reactivex.rxjava2:rxandroid:${rxAndroidVersion}",
-            rxJava    : "io.reactivex.rxjava2:rxjava:${rxJavaVersion}",
-            appSupport: "com.android.support:appcompat-v7:${appSupportVersion}",
-            kotlin    : "org.jetbrains.kotlin:kotlin-stdlib:${kotlin_version}"
+            rxAndroid : "io.reactivex.rxjava2:rxandroid:$rxAndroidVersion",
+            rxJava    : "io.reactivex.rxjava2:rxjava:$rxJavaVersion",
+            appSupport: 'androidx.appcompat:appcompat:1.0.0',
```
Comment (SartiMau) [inline buildsystem/dependencies.gradle]: 1-. please match the name with the dependency 
2-. make a variable for the version please

File: buildsystem/dependencies.gradle [hunk 9]
```diff
@@ -1,8 +1,8 @@
...
     //Android
-    androidMinSdkVersion = 15
-    androidTargetSdkVersion = 26
-    androidCompileSdkVersion = 26
+    androidMinSdkVersion = 22
+    androidTargetSdkVersion = 28
+    androidCompileSdkVersion = 28
 
     //General dependencies version
     appSupportVersion = '26.1.0'
```
Comment (SartiMau) [inline buildsystem/dependencies.gradle]: delete this, as it is not necessary

File: buildsystem/dependencies.gradle [hunk 10]
```diff
@@ -26,29 +26,29 @@ ext {
...
-            mockitoInline: "org.mockito:mockito-inline:${mockitoInlineVersion}",
+            mockitoInline: "org.mockito:mockito-inline:$mockitoInlineVersion",
     ]
 
     appDependencies = [
-            picasso         : "com.squareup.picasso:picasso:${picassoVersion}",
-            recyclerView    : "com.android.support:recyclerview-v7:${recyclerViewVersion}",
-            constraintLayout: "com.android.support.constraint:constraint-layout:${constraintLayoutVersion}"
+            picasso         : "com.squareup.picasso:picasso:$picassoVersion",
+            recyclerView    : 'androidx.recyclerview:recyclerview:1.0.0',
```
Comment (SartiMau) [inline buildsystem/dependencies.gradle]: make a variable for the version please

File: buildsystem/dependencies.gradle [hunk 11]
```diff
@@ -26,29 +26,29 @@ ext {
...
+            constraintLayout: 'androidx.constraintlayout:constraintlayout:1.1.3'
```
Comment (SartiMau) [inline buildsystem/dependencies.gradle]: make a variable for the version please

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (regmarcos) [reply app/build.gradle]: done

File: app/build.gradle (hunk 2 above)
Comment (regmarcos) [reply app/build.gradle]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt (hunk 3 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt (hunk 4 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt]: done

File: app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: buildsystem/dependencies.gradle (hunk 10 above)
Comment (regmarcos) [reply buildsystem/dependencies.gradle]: done

File: buildsystem/dependencies.gradle (hunk 11 above)
Comment (regmarcos) [reply buildsystem/dependencies.gradle]: done

File: buildsystem/dependencies.gradle (hunk 9 above)
Comment (regmarcos) [reply buildsystem/dependencies.gradle]: done

File: buildsystem/dependencies.gradle (hunk 8 above)
Comment (regmarcos) [reply buildsystem/dependencies.gradle]: done

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/res/values/colors.xml [hunk 12]
```diff
@@ -3,4 +3,5 @@
     <color name="colorPrimary">#3F51B5</color>
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="FAB_color">#353892</color>
```
Comment (SartiMau) [inline app/src/main/res/values/colors.xml]: resource keys must be in lowercase and with the word separated by _

Comment (regmarcos) [reply app/src/main/res/values/colors.xml]: done

**Review by SartiMau — APPROVED** [review]

**Review by Daniel-Juarez — APPROVED** [review]

**Pull request merged**